More detail about Project Director requirements (#86)
rust-lang/leadership-council: roles/rust-foundation-project-director.md

- A keen sense of the needs of Rust Project, its teams, processes, and ecosystem.
- Willingness to understand and represent viewpoints they may not have personal experience with or necessarily agree with.

> ## Requirements and Eligibility

Reviewer: I believe that there is also a requirement to be in "good standing" with the project. I think this doesn't have a formal definition, but loosely has meant something akin to not being under any kind of moderation sanction. It might be hard to pin down a definition that would be satisfying in legal contexts, but I don't think that's necessary so perhaps we should try to sketch something out?

Author: I added "Project directors must be members of the Rust Project in good standing" with some extra detail about moderation sanction.

I know we're leaving a certain amount of ambiguity with that, but I think this still provides more clarity and we can better define "good standing" if needed in the future.

There are requirements and eligibility criteria Project Directors must meet.
These are set by the Rust Project, the [Foundation bylaws], the laws of the State of Delaware, and the laws of the United States of America.

- Project directors must disclose their name on the Foundation's IRS Form 990, which is a public document. For an example, see [the Foundations 2021 filing][form-990].
- A candidate may not be elected as a Project Director if they are employed by a corporate member of the Foundation that already employs another Project Director.
  This comes from [Section 4.3(g)] of the Foundation bylaws.

>   In the past there has been confusion about the definition of employment.

Reviewer 2: This sentence feels kind of out of place.  I would prefer to state that a broad definition has been used in the past and then proceed to describe that definition and not say that there has been confusion.

Reviewer 3: :+1: and I'd also like to lay stress on the legal element here - it isn't a definition we've come up with in isolation. Delaware corporate law generally interprets 'employment' broadly, with the key element being that the individual is engaged to carry out work or services for the company. That can cover a range of different types of relationship.

Author: Sounds good. I deleted the line about past confusion and added one about Delaware corporate law.

I do think referencing Delaware law is important here, because it makes it clear that the Foundation isn't just pulling stricter definitions out of thin air.

  Thus far the Project and Foundation have used a fairly broad definition that includes contracting relationships.
- Project Directors must disclose any conflicts of interest to the Foundation.
  These disclosures are kept confidential by the Foundation and are not shared with the public.

## Term Length

- 2 year term

[director-role]: https://foundation.rust-lang.org/static/board-director-role-description.pdf
[Foundation bylaws]: https://foundation.rust-lang.org/policies/bylaws/#article-iv%3A-directors
[form-990]: https://foundation.rust-lang.org/static/publications/financial-filings/form-990-2021.pdf

> [Section 4.3(g)]: https://foundation.rust-lang.org/policies/bylaws/#section-4.3-nomination%2C-election-and-term-of-office-of-directors:~:text=No%20person%20may%20be%20elected%20Project%20Director%20if%2C%20at%20the%20time%20of%20election%2C%20that%20person%20is%20employed%20by%20a%20Corporate%20Member%20and%20that%20Corporate%20Member%20or%20any%20Related%20Company%20already%20employs%20a%20Project%20Director.

Reviewer 4: The [scroll to text fragment](https://caniuse.com/url-scroll-to-text-fragment) link doesn't work in Firefox. Would it be possible to just link to 4.3 without the `:~:text` part?

Author: Done!

- A keen sense of the needs of Rust Project, its teams, processes, and ecosystem.
- Willingness to understand and represent viewpoints they may not have personal experience with or necessarily agree with.

## Requirements and Eligibility

There are requirements and eligibility criteria Project Directors must meet.
These are set by the Rust Project, the [Foundation bylaws], the laws of the State of Delaware, and the laws of the United States of America.

- Project directors must be members of the Rust Project in good standing.
  Among other things, this means they must not be currently under any moderation sanction.
- Project directors must disclose their name on the Foundation's IRS Form 990, which is a public document. For an example, see [the Foundations 2021 filing][form-990].

> - A candidate may not be elected as a Project Director if they are employed by a corporate member of the Foundation that already employs another Project Director.

Reviewer 5: ```suggestion
- No two project directors can be employed by the same corporate member of the Foundation.
```

Maybe we can simplify this language to something like this? I think this is essentially the same language. I imagine that from a Rust project perspective, we may not care about the "corporate member" perspective -- we'd probably want this to just be "any company", but that can be discussed separately.

Author: The wording update seems reasonable to me.

My reading of the bylaws is that if some company was not a corporate member then all of the project directors could be from the same company. Is that corrent? If so, is that intended?

I could imagine we want to make it "any company" rather than "any corporate member," but if we do, I think we should make clear how much of that restriction is from the bylaws and how much is the Project adding their own requirements on top of that.
